networkit/cpp/graph/GraphBuilder.cpp

```diff
@@ -145,11 +145,11 @@ void GraphBuilder::increaseInWeight(node u, node v, edgeweight ew) {
 
 Graph GraphBuilder::toGraph(bool autoCompleteEdges, bool parallel) {
     Graph G(n, weighted, directed);
-    assert(G.outEdges.size() == n);
-    assert(G.outEdgeWeights.size() == (weighted ? n : 0));
-    assert(G.inEdges.size() == (directed ? n : 0));
-    assert(G.inEdgeWeights.size() == ((weighted && directed) ? n : 0));
-
+    #ifdef NETWORKIT_SANITY_CHECKS
+    assert(n == G.upperNodeIdBound());
+    assert(G.checkConsistency());
+    #endif
```

Comment:
The cheap `n == G.upperNodeIdBound()` assertion can stay outside of NETWORKIT_SANITY_CHECKS.